src/Analyzers/Core/Analyzers/AbstractCodeStyleDiagnosticAnalyzer.cs

```diff
@@ -55,6 +58,7 @@ protected AbstractCodeStyleDiagnosticAnalyzer(ImmutableArray<DiagnosticDescripto
                     DiagnosticSeverity.Hidden,
                     isEnabledByDefault: true,
                     description: description,
+                    helpLinkUri: $"https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/{id}",
```

Comment:
What about the rules that are documented together?
The link for them takes the form of  `https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/idexxxx-ideyyyy`

Reply:
> What about the rules that are documented together?

We have redirects for these, I verified this will work after my latest open PR is merged.

> I'd use the F1 link here.

We considered it, but it does not work for all rule IDs. Also confirmed with @gewarren that we should not rely on msdn redirections for help links.

For context, if I type out https://msdn.microsoft.com/query/roslyn.query?appId=roslyn&k=k(CA2000) it seems to redirect properly. Doesn't seem to work for all IDs though, for example https://msdn.microsoft.com/query/roslyn.query?appId=roslyn&k=k(CA1000) took me to the index (not sure if that is only for the first rule though). It doesn't work even for some rule in middle https://msdn.microsoft.com/query/roslyn.query?appId=roslyn&k=k(CA2234).

Reply:
> We have redirects for these, I verified this will work after my latest open PR is merged.

Great. There is no problem then if the redirections for these will always be maintained for future rules. 🎉 

> For context, if I type out https://msdn.microsoft.com/query/roslyn.query?appId=roslyn&k=k(CA2000) it seems to redirect properly. Doesn't seem to work for all IDs though, for example https://msdn.microsoft.com/query/roslyn.query?appId=roslyn&k=k(CA1000) took me to the index (not sure if that is only for the first rule though). It doesn't work even for some rule in middle https://msdn.microsoft.com/query/roslyn.query?appId=roslyn&k=k(CA2234).

Probably the F1 service redirections aren't working because these f1_keywords existed in VS docs. The service seems to have some cache of where each keyword exists. So it should work properly after some time. (This isn't to say we should rely on F1 service, just putting my thought on why it might be not working properly)